controllers/etcd_controller.go

```diff
@@ -670,12 +683,25 @@ func (r *EtcdReconciler) syncStatefulSetSpec(ss *appsv1.StatefulSet, cm *corev1.
 		err = nil
 	}
 	if err != nil {
-		logger.Infof("patching statefulset failed for %s", ss.Name)
 		return nil, err
 	}
 	return ssCopy, err
 }
 
+func (r *EtcdReconciler) recreateStatefulset(ss *appsv1.StatefulSet) error {
+
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
Please merge this `RetryOnConflict` with the one [below](https://github.com/gardener/etcd-druid/pull/33/files#diff-8e5c816f8b2f1adeef36a64d81b3e0f6R699).

Reply:
For the case where delete succeeded but create failed for some reason, you could save the info in a variable.

For example,
```go
func (r *EtcdReconciler) recreateStatefulset(ss *appsv1.StatefulSet) error {
        skipDelete := false
	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
                 if !skipDelete {
		        if err := r.Delete(context.TODO(), ss); err != nil && !errors.IsNotFound(err) {
			        return err
		        }
                 }

                 skipDelete = true

		return r.Create(context.TODO(), ss)
	})
	return err
}
```